// angular-bootstrap-md/esm2015/lib/free/collapse/collapse.module.js
import * as ɵngcc0 from '@angular/core';
var CollapseModule_1;
import { __decorate } from "tslib";
import { NgModule } from '@angular/core';
import { CollapseComponent } from './collapse.component';
let CollapseModule = CollapseModule_1 = class CollapseModule {
    static forRoot() {
        return { ngModule: CollapseModule_1, providers: [] };
    }
};
CollapseModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: CollapseModule });
CollapseModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function CollapseModule_Factory(t) { return new (t || CollapseModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(CollapseModule, { declarations: function () { return [CollapseComponent]; }, exports: function () { return [CollapseComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CollapseModule, [{
        type: NgModule,
        args: [{
                declarations: [CollapseComponent],
                exports: [CollapseComponent]
            }]
    }], null, null); })();
export { CollapseModule };
