// angular-bootstrap-md/esm2015/lib/free/dropdown/dropdown.state.js
import { __decorate, __metadata } from "tslib";
import { EventEmitter, Injectable } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
let BsDropdownState = class BsDropdownState {
    constructor() {
        this.direction = 'down';
        this.isOpenChange = new EventEmitter();
        this.isDisabledChange = new EventEmitter();
        this.toggleClick = new EventEmitter();
        this.dropdownMenu = new Promise((resolve) => {
            this.resolveDropdownMenu = resolve;
        });
    }
};
BsDropdownState.ɵfac = function BsDropdownState_Factory(t) { return new (t || BsDropdownState)(); };
BsDropdownState.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: BsDropdownState, factory: function (t) { return BsDropdownState.ɵfac(t); } });
BsDropdownState = __decorate([ __metadata("design:paramtypes", [])
], BsDropdownState);
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(BsDropdownState, [{
        type: Injectable
    }], function () { return []; }, null); })();
export { BsDropdownState };
