// angular-bootstrap-md/esm2015/lib/free/waves/waves.module.js
import * as ɵngcc0 from '@angular/core';
var WavesModule_1;
import { __decorate } from "tslib";
import { NgModule } from '@angular/core';
import { WavesDirective } from './waves-effect.directive';
let WavesModule = WavesModule_1 = class WavesModule {
    static forRoot() {
        return { ngModule: WavesModule_1, providers: [] };
    }
};
WavesModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: WavesModule });
WavesModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function WavesModule_Factory(t) { return new (t || WavesModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(WavesModule, { declarations: function () { return [WavesDirective]; }, exports: function () { return [WavesDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(WavesModule, [{
        type: NgModule,
        args: [{
                declarations: [WavesDirective],
                exports: [WavesDirective]
            }]
    }], null, null); })();
export { WavesModule };
